src/osyris/spatial/subdomain.py

```diff
@@ -25,10 +26,14 @@ def extract_sphere(dataset, radius, origin):
         if np.any(c):
             subdomain[name] = dataset[name][c]
 
+    change_origin(subdomain, origin)
+    if basis is not None:
+        change_basis(subdomain, basis, dr_L)
+
     return subdomain
 
 
-def extract_box(dataset, dx, dy, dz, origin):
+def extract_box(dataset, dx, dy, dz, origin, basis=None, dr_L=None):
```

Comment:
Thinking about this again, I'm wondering if it would be more useful with `xmin, xmax...` syntax. Say I would like to get all positive x values, all i need is to say `xmin = 0` and no need to specify other aguments, instead of having to find the center at `0.25` and then `dx=0.5`.

Reply:
I prefer if we keep dxyz since it's shorter than providing xyz min/max, but I can change it since I don't mind that much.